### PR TITLE
Fast mode in SCALrecommendation and in interpolation

### PR DESCRIPTION
**Pull request by olelod**
Adding fast mode to SCALrecommendation as well, and to the interpolation module.

### CODE REVIEW

File: pyscal/scalrecommendation.py [hunk 1]
```diff
@@ -81,6 +87,7 @@ def __init__(self, low, base, high, tag=None, h=0.01):
                 swcr=low["swcr"],
                 h=h,
                 tag=tag,
+                fast=fast,
```
Comment (berland) [inline pyscal/scalrecommendation.py:90]: These code lines are deprecated and will be removed, see #228

Comment (olelod) [reply pyscal/scalrecommendation.py:90]: Ah. Did not catch that one, sorry. So the fast variable will not really be needed in the constructor here at all then, only in the interpolate function?

**Comment by berland** [comment]
Recommend to start with tests, adding 
``` 
interpolant = scalrec.interpolate(-0.5)
assert interpolant.fast
```
to `test_factory.py::test_fast_mode()` proves the deficiency you found

File: pyscal/scalrecommendation.py (hunk 1 above)
Comment (berland) [reply pyscal/scalrecommendation.py]: Just leave out 'fast=fast' for the situation when the class input is dicts, then it will not conflict with #228..

**Comment by berland** [comment]
Recommend doing `pre-commit install` in your cloned copy to avoid commiting non-blackened files.

File: pyscal/scalrecommendation.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ class SCALrecommendation(object):
         tag (str): A string that describes the recommendation. Optional.
     """
 
-    def __init__(self, low, base, high, tag=None, h=0.01):
+    def __init__(self, low, base, high, tag=None, h=0.01, fast=False):
```
Comment (berland) [inline pyscal/scalrecommendation.py:28]: This should not really be necessary, as it should be assumed given  in the `low`, `base` and `high` objects. 

The class must decide what to do if `low.fast != base.fast` for example.

Or,  do we need to support overriding `low.fast`?

File: pyscal/scalrecommendation.py [hunk 3]
```diff
@@ -289,14 +293,14 @@ def interpolate(self, parameter, parameter2=None, h=0.02):
             )
         tagstring = "\n".join(tags)
         if do_gaswater:
-            interpolant = GasWater(h=h, tag=tagstring)
+            interpolant = GasWater(h=h, tag=tagstring, fast=self.fast)
```
Comment (berland) [inline pyscal/scalrecommendation.py:296]: inherit fast setting from interpolants?

File: pyscal/scalrecommendation.py (hunk 2 above)
Comment (olelod) [reply pyscal/scalrecommendation.py:28]: I guess all should be .fast for fast-mode to be activated, but maybe just issue a warning not an error if low.fast != base.fast.

File: tests/test_pyscallist.py [hunk 4]
```diff
@@ -699,3 +699,38 @@ def test_error_messages_pr_satnum(caplog):
...
+    dframe = pd.DataFrame(
+        columns=["SATNUM", "Nw", "Now", "ng", "nog"],
+        data=[
+            [1, 2, 2, 2, 2],
+            [2, 2, 2, 2, 2],
+            [3, 2, 2, 2, 2],
+        ],
+    )
+    relperm_data = PyscalFactory.load_relperm_df(dframe)
+    p_list_fast = PyscalFactory.create_pyscal_list(relperm_data, h=0.2, fast=True)
```
Comment (berland) [inline tests/test_pyscallist.py:734]: Maybe similar test for `gasoil_list`, `wateroil_list`, `gaswater_list` ?

**Comment by berland** [comment]
Can you check what happens if "fast" is present as a column in input dataframes when given to pyscal list factory methods? The dataframe rows are treated as dictionaries, and passed around as ** arguments to python function, which might case bugs?

Users might guess that fast-mode is individually adjustable through the dataframe input, while we maybe don't want to support that. Should we warn if that parameter is found?

**Comment by olelod** [comment]
> Can you check what happens if "fast" is present as a column in input dataframes when given to pyscal list factory methods? The dataframe rows are treated as dictionaries, and passed around as ** arguments to python function, which might case bugs?
> 
> Users might guess that fast-mode is individually adjustable through the dataframe input, while we maybe don't want to support that. Should we warn if that parameter is found?

I'll do a check. I was thinking about the option of making "fast" a column in the input dataframe, but I could not really see why a user would want to do that. I guess it would be good to issue a warning stating it is a "global" option.

**Comment by olelod** [comment]
Nothing happens when adding fast as a column in the dataframe. Added a warning to `load_relperm_df` that is triggered if a fast column is present.

File: tests/test_pyscallist.py [hunk 5]
```diff
@@ -699,3 +699,75 @@ def test_error_messages_pr_satnum(caplog):
...
+        assert item.fast
+
+    # GasWater list
+    input_dframe = dframe[["SATNUM", "nw", "ng"]].copy()
+    relperm_data = PyscalFactory.load_relperm_df(input_dframe)
+    p_list_fast = PyscalFactory.create_pyscal_list(relperm_data, h=0.2, fast=True)
+    for item in p_list_fast:
+        assert item.fast
+
+    # Testing with fast column in dataframe
```
Comment (berland) [inline tests/test_pyscallist.py:760]: Make  a comment for future test code maintainer that this is intentioanlly not supported?

File: pyscal/factory.py [hunk 6]
```diff
@@ -755,6 +754,13 @@ def load_relperm_df(inputfile, sheet_name=None):
             "SATNUM " + input_df["SATNUM"].astype(str) + " " + input_df["TAG"]
         )
 
+        # Check if fast is defined as a column
+        if "fast" in input_df:
+            logger.warning("Fast mode is not an option for individual SATNUMs")
+            logger.warning("it is implemented as a global option.")
+            logger.warning("The fast column in the dataframe will be ignored.")
+            logger.warning("Use fast=True in the function call instead.")
```
Comment (berland) [inline pyscal/factory.py:762]: Command line client users will see this if they expore possibilities for setting fast mode. Say in the last line that fast mode is only available through Python API?

**Review by berland — APPROVED** [review]
Looks good!

Rewrite the commit message when squashing.

**Pull request merged**